cmd/landscaper-controller/app/options.go

```diff
@@ -71,27 +71,29 @@ func (o *Options) Complete(ctx context.Context) error {
 
 func (o *Options) parseConfigurationFile(ctx context.Context) (*config.LandscaperConfiguration, error) {
 	decoder := serializer.NewCodecFactory(api.ConfigScheme).UniversalDecoder()
+
+	configv1alpha1 := &v1alpha1.LandscaperConfiguration{}
+	api.ConfigScheme.Default(configv1alpha1)
+	config := &config.LandscaperConfiguration{}
+	err := api.ConfigScheme.Convert(configv1alpha1, config, ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	if len(o.ConfigPath) == 0 {
-		configv1alpha1 := &v1alpha1.LandscaperConfiguration{}
-		api.ConfigScheme.Default(configv1alpha1)
-		config := &config.LandscaperConfiguration{}
-		err := api.ConfigScheme.Convert(configv1alpha1, config, ctx)
-		if err != nil {
-			return nil, err
-		}
 		return config, nil
 	}
+
 	data, err := ioutil.ReadFile(o.ConfigPath)
 	if err != nil {
 		return nil, err
 	}
 
-	cfg := &config.LandscaperConfiguration{}
-	if _, _, err := decoder.Decode(data, nil, cfg); err != nil {
+	if _, _, err := decoder.Decode(data, nil, config); err != nil {
```

Comment:
I think the changes will not be merged but overwritten?
you could do, similar to above, first decode as v1alpha, then default and then convert to the core config.

Reply:
I have debugged this. The default config contains for example the ls-system namespace twice in the deployerManagement:
```yaml
deployerManagement:
  namespace: ls-system
  agent:
    namespace: ls-system
```
Then `Decode` is applied with a custom config which contains only the second namespace:
```yaml
deployerManagement:
  agent:
    namespace: lndscpr-int-test
```
The result is a mixture of both:
```yaml
deployerManagement:
  namespace: ls-system
  agent:
    namespace: lndscpr-int-test
```
So, the fields of the default config which are not overwritten by the custom config seem to survive.

Reply:
hmm ok they survive but I think this is more due to weird internal struct overwrite.
I think this could lead to unintended behavior in the future (especially for slices or maps).
Therefore I would prefer to keep it clean and explicitly run the defaulting code.

Reply:
I have now changed it how you suggested. That's the better way.